File: C++/Rank Up/Is a number prime?.cpp
#include <iostream>
#include <cmath>

// Returns "true" if 'num' is a prime number
bool isPrime(int const &num)
{
    if (num <= 1)
        return false;

    for (int i{2}; i <= static_cast<int>(std::sqrt(num)); i++)
    {
        if (num % i == 0)
            return false;
    }
    return true;
}

int main()
{
    std::cout << isPrime(5) << '\t' << isPrime(4483) << '\t' << isPrime(3164)
              << '\t' << isPrime(7789) << '\t' << isPrime(1) << '\t' << isPrime(0)
              << '\t' << isPrime(4) << std::endl;

    return 0;
}

File: Python/Rank Up/Is a number prime?.py
import math


def is_prime(num: int) -> bool:
    """ Returns True if 'num' is prime. Otherwise - False """
    if num <= 1:
        return False

    i = 2
    while i <= int(math.sqrt(num)):
        if num % i == 0:
            return False
        i += 1
    return True


print(is_prime(5), is_prime(4483), is_prime(3164),
      is_prime(7789), is_prime(1), is_prime(0))
